README.adoc

```diff
@@ -13,30 +29,273 @@ These can be installed via the provided `.tool-versions` using `asdf install`.
 ----
 
 == Development Process
 
-[source,shell]
+[source, shell]
 ----
-yarn dev
+NODE_TLS_REJECT_UNAUTHORIZED=0 yarn dev
 ----
 
-If you are working with a local Connect, you need to run:
+If you want to build and start the app, you should run the following:
 
-[source,shell]
+[source, shell]
 ----
-NODE_TLS_REJECT_UNAUTHORIZED=0 yarn dev
+yarn build && yarn start -p $CONNECT_ACCOUNT_PORT
+----
+
+The `yarn start` port is not set by default, as it is used by Vercel for building the app.
+
+== Testing
+
+[source, shell]
+----
+yarn test
+----
+
+== Environment variables
+
+|===
+| Name | Description
+
+| `CONNECT_ACCOUNT_PORT`
+| Local port used to run the application.
+
+| `CONNECT_ACCOUNT_SESSION_SALT`
+| The password used to seal or access the cookie session. It needs to be at least 32 characters long. 
+
+| `CONNECT_MANAGEMENT_URL`
+| URL used to fetch identities from the management GraphQL endpoint.
+
+| `CONNECT_MANAGEMENT_API_KEY`
+| API key used to access the management GraphQL endpoint.
+
+| `CONNECT_PROVIDER_URL`
+| URL used to start the connect oauth flow.
+
+| `CONNECT_APPLICATION_CLIENT_ID`
+| Client ID of the online service (e.g. internet website, application) that uses the Provider Authentication and Authorization service for its User.
+
+| `CONNECT_APPLICATION_CLIENT_SECRET`
+| Paired with the client ID, used to authenticate the Application from which the User intent to sign in.
+
+| `CONNECT_APPLICATION_SCOPES`
+| Represents the kind of user authorized information and actions that an Application is able to access on another Application.
+
+| `NEXT_PUBLIC_SENTRY_DSN`
+| Data Source Name representing the configuration required by the Sentry SDKs.
+|===
+
+
+== Connect login implementation
+
+To understand the flow of `connect-account`, you should read the `connect` https://developers.fewlines.co/guides/authorization_code_flow/overview/[documentation].
+
+[source, jsx]
 ----
+<BusinessComponent>
+  {({ businessLogic }) => (
+    <VisualizationComponent businessLogic={businessLogic} />
+  )}
+</BusinessComponent>
+----
+
 
 == Source Code Organization
+
+We are using the NextJS folder architecture (i.e. `/pages`) to utilize its router, out of the box. For more information, please refer to the https://nextjs.org/docs/basic-features/pages[documentation].
+
+We are also using the `Command Query Responsibility Segregation`(CQRS) pattern to separate queries from mutations. They are located in the `queries/` and `command/` folder.
+
+=== /.github
+
+- */workflows*: GitHub Actions used to run tests during CI/CD process flow.
+- *PULL_REQUEST_TEMPLATE*: Template used when opening a pull request on GitHub.
+
+=== /src
+
+- *@types/*: Type declaration used in multiple places.
+- *assets/*: Folder containing assets.
+- *commands/*: Write (e.g. `POST`) database actions.
+- *components/*:
+  - *business/*: React components that provide business logic through render props. You shouldn't touch them.
+  - *visualization/*: React component that  
```

Comment:
that?

Reply:
> React component that only — for the most part — handle the visualization part. 

Not a huge fan but I can't really find better wording.

Reply:
👍